diffusion-model-with-mnist-data.py

```diff
@@ -167,3 +167,53 @@ def forward(self, noisy_images, timestep):
 # # Evaluation
 
 
+# -
+
+# # Sampling
+
+# +
+def sample_image(image_size, beta, noise_predictor, simple_variance=False):
+    "sample image"
+    
+    timesteps = beta.shape[0] - 1
+    alpha = 1 - beta
+    alpha_cum = torch.cumprod(alpha, dim=0)
+    if simple_variance:
+        variances = beta
+    else:
+        alpha_cum_t = torch.cat(alpha_cum, torch.Tensor([0]))
+        alpha_cum_t_minus_1 = torch.cat(torch.Tensor([0]), alpha_cum)
+        variances = (1-alpha_cum_t_minus_1)/(1-alpha_cum_t)
```

Comment:
Use 
```Python
alpha_cum_t = alpha_cum
alpha_cum_t_minus_1 = torch.cat(torch.Tensor([0]), alpha_cum)
variances = (1-alpha_cum_t_minus_1)/(1-alpha_cum_t) * beta
```

(No need to expand alpha_cum_t)

Reply:
how else can i make vector-operations including alpha_t and alpha_t-1?

Reply:
Please elaborate. The code snippet I posted should work fine.